ADAL/src/utils/NSString+ADHelperMethods.m

```diff
@@ -286,6 +286,27 @@ + (BOOL)adIsStringNilOrBlank:(NSString *)string
     return [string rangeOfCharacterFromSet:nonWhiteCharSet].location == NSNotFound;
 }
 
+- (long)adFindCharactersFromSet:(NSCharacterSet *)set
+                           start:(long)startIndex
+{
+    THROW_ON_NIL_ARGUMENT(set);
+    long end = self.length;
+    if (startIndex >= end)
+        return end;
+    
+    NSRange toSearch = {.location  = startIndex, .length = (end - startIndex)};
+    long found = [self rangeOfCharacterFromSet:set options:NSLiteralSearch range:toSearch].location;
+    return (found == NSNotFound) ? end : found;
```

Comment:
I'm curious about the contract here, and how come it'd be different from what rangeOfCharacterFromSet would return. i.e. why would you want to return self.length instead of NSNotFound if you can't find the character?